torch/csrc/jit/script/compiler.cpp

```diff
@@ -555,12 +563,21 @@ std::shared_ptr<SugaredValue> emitBuiltinCall(
 
   const auto& variants = getAllOperatorsFor(Symbol::aten(name));
   std::stringstream failure_messages;
-  for (const std::shared_ptr<Operator>& op : variants) {
-    if (auto result = tryEmitBuiltin(
-            op, failure_messages, loc, method, name, inputs, attributes)) {
-      return result;
+
+  //first we try to match the schema without any conversion
+  //if no schema matches then insert TensorToNum nodes where needed
+  for(bool convert_tensors_to_nums : {false, true}) {
+    for (const std::shared_ptr<Operator>& op : variants) {
+      //clear previous error messages
+      failure_messages.str("");
```

Comment:
Failure messages need to accumulate for _all_ of the options. This changes it to clear after each variant, which will lead to very misleading errors that only report problems for one overload. We should clear the failure_messages between convert_tensors_to_nums loops, not between variants.